packages/eslint-plugin/tests/docs.test.ts

```diff
@@ -0,0 +1,141 @@
+import fs from 'fs';
+import path from 'path';
+
+import marked from 'marked';
+import rules from '../src/rules';
+
+const docsRoot = path.resolve(__dirname, '../docs/rules');
+const rulesData = Object.entries(rules);
+
+function createRuleLink(ruleName: string): string {
+  return `[\`@typescript-eslint/${ruleName}\`](./docs/rules/${ruleName}.md)`;
+}
+
+function parseReadme(): marked.Tokens.Table {
+  const readmeRaw = fs.readFileSync(
+    path.resolve(__dirname, '../README.md'),
+    'utf8',
+  );
+  const readme = marked.lexer(readmeRaw, {
+    gfm: true,
+    silent: false,
+  });
+
+  // find the table
+  const rulesTable = readme.find(
+    token => token.type === 'table',
+  ) as marked.Tokens.Table;
```

Comment:
Nit - I probably should have correctly typed this

```suggestion
  ) as marked.Tokens.Table | undefined;
```

Reply:
that's correct, i just copy/paste it from tools :>

Reply:
this actually should be:
```suggestion
const rulesTable = readme.find(
    (token): token is marked.Tokens.Table => token.type === 'table',
);
```

Reply:
this actually should be:
```suggestion
const rulesTable = readme.find(
    (token): token is marked.Tokens.Table => token.type === 'table',
);
```